mautic/mautic
# Fix the dynamic content events are not displayed on the lead timeline

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes [#9597](https://github.com/mautic/mautic/issues/9597) 



#### Description:
We have faced with the same problem as described in the closed issue [#9597](https://github.com/mautic/mautic/issues/9597). 

I applied recommendations from the issue and checked it all. It makes sense and fixes the problem.


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new segment.
3. Create two dynamic content elements.
4. Create a new campaign with the segment from the step 3.
5. Add the `request dynamic content` decision connected with the first DC.
6. Add the `push dynamic content` action connected with the second DC.
7. Create a new standalone form with only an email address field and add an actions puts the leads into the segment from the step 2.
8. Create a (Mautic) landing page, put the form and the dynamic content `div` on it with the div name.
```
<div data-slot="dwc" data-param-slot-name="myslot">
  <h1>Dynamic web content for myslot</h1>
</div>

myslot is Requested Slot Name
```
10. Open the landing page in incognito mode.
11. Fill in and submit the form.
13. Update and trigger the campaign.
14. Reload the landing page.
15. Open the created contact details page.
16. The contact details page opens. This page hasn't opened before.
